src/docfx/build/xref/ExternalXrefMapLoader.cs

```diff
@@ -14,11 +14,42 @@ internal class ExternalXrefMapLoader
     {
         private static readonly byte[] s_uidBytes = Encoding.UTF8.GetBytes("uid");
 
-        public static IReadOnlyDictionary<string, Lazy<ExternalXrefSpec>> Load(Config config, FileResolver fileResolver)
+        public static IReadOnlyDictionary<string, Lazy<ExternalXrefSpec>> Load(
+            Config config,
+            FileResolver fileResolver,
+            Docset fallbackDocset,
+            Input input)
         {
             var result = new Dictionary<string, Lazy<ExternalXrefSpec>>();
 
             foreach (var url in config.Xref)
+            {
+                try
+                {
+                    LoadXref(url);
+                }
+                catch (DocfxException ex) when (ex.Error.Code == "file-not-found")
+                {
+                    FilePath path;
+                    if (fallbackDocset != null && input.Exists(path = new FilePath(url, FileOrigin.Fallback)))
+                    {
+                        using var stream = input.ReadStream(path);
+                        var xrefMap = JsonUtility.Deserialize<XrefMapModel>(new StreamReader(stream), path);
```

Comment:
>var xrefMap = JsonUtility.Deserialize<XrefMapModel>(new StreamReader(stream), path); [](start = 24, length = 84)

Are we assuming that xref map file in fallback repo is in json format?
(I'm not quite familiar with the context)

Reply:
you are right, it can be other formats